sympy/functions/elementary/piecewise.py

```diff
@@ -276,6 +277,18 @@ def eval(cls, *_args):
 
             current_cond.add(cond)
 
+            #   Does not add intervals which are already covered entirely by 
+            #   previous intervals (in terms of ordering of arguments)
+            #   Piecewise((0, t>=4), (0, t<3), (1, t<=2), (3, t<4))
+            #   >>> Piecewise((0, (t >= 4) | (t < 3)), (3, True))
+            from sympy import Intersection
+            cu_inter = cond.as_set()
```

Comment:
multivariate inequalities will raise NotImplementedError

Reply:
Hi @smichr I have made sure it does not happen now. Thanks for your suggestion!